File: native/chat-widget-android/chat-widget/src/main/java/com/bootdesk/chatwidget/BridgeConfig.kt
package com.bootdesk.chatwidget

enum class ThemeMode {
    AUTO,
    LIGHT,
    DARK,
}

data class ThemeConfig(
    val mode: ThemeMode? = null,
)

data class BridgeConfig(
    val title: String? = null,
    val locale: String? = null,
    val placeholder: String? = null,
    val theme: ThemeConfig? = null,
)

File: native/chat-widget-ios/Sources/BootdeskChatWidget/BridgeConfig.swift
import Foundation

public enum ThemeMode: String {
    case auto
    case light
    case dark
}

public struct ThemeConfig {
    public var mode: ThemeMode?

    public init(mode: ThemeMode? = nil) {
        self.mode = mode
    }
}

public struct BridgeConfig {
    public var title: String?
    public var locale: String?
    public var placeholder: String?
    public var theme: ThemeConfig?

    public init(
        title: String? = nil,
        locale: String? = nil,
        placeholder: String? = nil,
        theme: ThemeConfig? = nil
    ) {
        self.title = title
        self.locale = locale
        self.placeholder = placeholder
        self.theme = theme
    }
}
